File: aemet/SergioDeIscar-aemet-java/SergioDeIscar-aemet-java/src/main/java/org/example/controllers/InformeController.java
package org.example.controllers;

import org.example.models.Informe;
import org.example.repositories.informe.InformeExtension;
import org.example.repositories.informe.InformeRepository;

import java.time.LocalDate;
import java.util.List;

public class InformeController implements InformeExtension {
    private final InformeRepository repo;
    public InformeController(InformeRepository repo) {
        this.repo = repo;
    }
    @Override
    public List<Informe> getAll() {
        return repo.getAll();
    }

    @Override
    public Informe getById(LocalDate localDate) {
        return repo.getById(localDate);
    }

    @Override
    public Informe save(Informe element, boolean storage) {
        return repo.save(element, storage);
    }

    @Override
    public void saveAll(List<Informe> elements, boolean storage) {
        repo.saveAll(elements, storage);
    }

    @Override
    public Informe deleteById(LocalDate localDate) {
        return repo.deleteById(localDate);
    }

    @Override
    public Informe update(Informe element) {
        return repo.update(element);
    }

    @Override
    public Informe updateById(LocalDate localDate, Informe element) {
        return repo.updateById(localDate, element);
    }
}

File: aemet/SergioDeIscar-aemet-kotlin/SergioDeIscar-aemet-kotlin/src/main/kotlin/controllers/InformeController.kt
package controllers

import models.Informe
import mu.KotlinLogging
import repositories.informe.InformeExtension
import repositories.informe.InformeRepository
import java.time.LocalDate

private val logger = KotlinLogging.logger {  }

class InformeController(
    private val informeRepository: InformeRepository
): InformeExtension {
    override fun getAll(): List<Informe> {
        logger.debug { "InformeController ->\tgetAll" }
        return informeRepository.getAll()
    }

    override fun getById(id: LocalDate): Informe? {
        logger.debug { "InformeController ->\tgetById" }
        return informeRepository.getById(id)
    }

    override fun save(element: Informe, storage: Boolean): Informe {
        logger.debug { "InformeController ->\tsave" }
        return informeRepository.save(element, storage)
    }

    override fun saveAll(elements: List<Informe>, storage: Boolean) {
        logger.debug { "InformeController ->\tsaveAll" }
        informeRepository.saveAll(elements, storage)
    }

    override fun deleteById(id: LocalDate): Informe? {
        logger.debug { "InformeController ->\tdeleteById" }
        return informeRepository.deleteById(id)
    }

    override fun update(element: Informe): Informe? {
        logger.debug { "InformeController ->\tupdate" }
        return informeRepository.update(element)
    }

    override fun updateById(id: LocalDate, element: Informe): Informe? {
        logger.debug { "InformeController ->\tupdateById" }
        return informeRepository.updateById(id, element)
    }
}